cum/scrapers/mangadex.py

```diff
@@ -0,0 +1,143 @@
+from bs4 import BeautifulSoup
+from cum import config, exceptions, output
+from cum.scrapers.base import BaseChapter, BaseSeries, download_pool
+from functools import partial
+from mimetypes import guess_type
+import concurrent.futures
+import re
+import requests
+
+class MangadexSeries(BaseSeries):
+    url_re = re.compile(r'(?:https?://mangadex\.com)?/manga/([0-9]+)')
+    # Example chapter-name inputs:
+    # Vol. 2 Ch. 18 - Strange-flavored Ramen
+    # Ch. 7 - Read Online
+    # Vol. 01 Ch. 001-013 - Read Online
+    # Oneshot
+    name_re = re.compile(r'Ch\. ?([A-Za-z0-9\.\-]*)(?: - (.*))')
+    language_re = re.compile(r'/images/flags/')
+    group_re =  re.compile(r'/group/([0-9]+)')
+
+    def __init__(self, url, **kwargs):
+        super().__init__(url, **kwargs)
+        self._get_page()
+        self.chapters = self.get_chapters()
+
+    def _get_page(self):
+        r = requests.get(self.url)
+        self.soup = BeautifulSoup(r.text, config.get().html_parser)
+
+    def get_chapters(self):
+        links = self.soup.find_all('a')
+        chapters = []
+
+        manga_name = self.name
+        for a in links:
+            url = a.get('href')
+            url_match = re.search(MangadexChapter.url_re, url)
+            if not url_match:
+                continue
+            name = a.string
+            name = name.strip()
+            name_parts = re.search(self.name_re, name)
+            chapter = name_parts.group(1) if name_parts else name
+            title = name_parts.group(2) if name_parts else None
+            title = None if title == "Read Online" else title
+            # TODO add an option to filter by language
```

Comment:
Before there is an option to filter out by language, we should hardcode it to English to match the (soon removed) Batoto scraper, since people probably do not expect to get Italian releases out of cum at the moment.